File: Arrays: Carry Forward/leaders.py
def countLeaders(A):
    n = len(A)
    c = 1
    maxVal = A[n - 1]

    for i in range(n - 2, -1, -1):
        if A[i] > maxVal:
            c += 1
            maxVal = A[i]

    return c

File: Arrays: Carry Forward/Leaders.java
public class Leaders {
    public int solve(int[] A) {
        int n = A.length;
        int c = 1;
        int maxVal = A[n-1];
        
        for (int i = n - 2; i >= 0; i--) {
            if (A[i] > maxVal) {
                c++;
                maxVal = A[i];
            }
        }
        
        return c;
    }

    public static void main(String[] args) {
        
    }
}
